UnicornEngine/video/source/vulkan/Renderer.cpp

```diff
@@ -1208,6 +1387,70 @@ bool Renderer::CreateSemaphores()
     return false;
 }
 
+bool Renderer::CreatePipelineCache()
+{
+    vk::PipelineCacheCreateInfo pipelineCacheCreateInfo;
+    if(m_vkLogicalDevice.createPipelineCache(&pipelineCacheCreateInfo, nullptr, &pipelineCache) != vk::Result::eSuccess)
+    {
+        LOG_ERROR( "Can't create pipeline cache!" );
+        return false;
+    }
+    return true;
+}
+
+bool Renderer::LoadEngineHelpData()
+{
+    unicorn::video::Texture texture;
+    std::string path = "data/textures/replace_me.jpg";
+    if(!texture.Load(path))
+    {
+        LOG_ERROR( "Can't find texture with path - %s", path.c_str() );
+        return false;
+    }
+    VkTexture* replaceMeTexture = new VkTexture(m_vkLogicalDevice);
+
+    if(!replaceMeTexture->Create(m_vkPhysicalDevice, m_vkLogicalDevice, m_commandPool, m_graphicsQueue, &texture))
+    {
+        LOG_ERROR("Can't create 'replace me' texture - %s", path.c_str());
+        return false;
+    }
+
+    m_replaceMeTextureHandle = texture.GetId();
+
+    vk::DescriptorSetAllocateInfo allocInfo;
+    allocInfo.descriptorPool = m_descriptorPool;
+    allocInfo.descriptorSetCount = 1;
+    allocInfo.pSetLayouts = &m_descriptorSetLayouts[1];
+
+    vk::DescriptorSet descriptorSet;
+
+    auto result = m_vkLogicalDevice.allocateDescriptorSets(&allocInfo, &descriptorSet);
+
+    if(result != vk::Result::eSuccess)
+    {
+        LOG_ERROR("Can't allocate descriptor sets!");
+        return false;
+    }
+
+    vk::WriteDescriptorSet imageDescriptorSet;
+    imageDescriptorSet.setDstSet(descriptorSet);
+    imageDescriptorSet.setDescriptorType(vk::DescriptorType::eCombinedImageSampler);
+    imageDescriptorSet.setDescriptorCount(1);
+    imageDescriptorSet.setPImageInfo(&replaceMeTexture->GetDescriptorImageInfo());
+
+    m_vkLogicalDevice.updateDescriptorSets(1, &imageDescriptorSet, 0, nullptr);
+
+    VkMaterial replaceMeMaterial;
+    replaceMeMaterial.texture = replaceMeTexture;
```

Comment:
Who is responsible to delete this texture?

Reply:
look at 630 - void Renderer::FreeTextures()